File: chapter3 - Stacks and Queues/3.5_sortStack/JavaScript/Main.js
const Stack = require("./Stack");
const SortStack = require("./SortStack");

const stack = new Stack();

stack.push(5);
stack.push(4);
stack.push(3);
stack.push(2);
stack.push(1);

const sortStack = new SortStack();

sortStack.sort(stack);

// Tests

console.log(stack.pop(), 5);
console.log(stack.pop(), 4);
console.log(stack.pop(), 3);
console.log(stack.pop(), 2);
console.log(stack.pop(), 1);

File: chapter3 - Stacks and Queues/3.5_sortStack/Java/Main.java
import java.util.Stack;

class Main {
  public static void main(String args[]) {
    SortStack sortStack = new SortStack();
    Stack<Integer> stack = new Stack<Integer>();

    stack.push(99);
    stack.push(4);
    stack.push(1);
    stack.push(6);
    stack.push(8);
    stack.push(10);
    stack.push(22);
    stack.push(3);
    stack.push(72);

    sortStack.sort(stack);

    while (!stack.isEmpty()) {
      System.out.println(stack.pop());
    }
  }
}